### PR TITLE
Add test suite along with support for Webpack v3

### PR DESCRIPTION
**Pull request by fny**
This pull requests adds what I'd say is a pretty good start for a test suite along with some small tweaks to support Webpack v3. This hopefully closes both #25 and #40!

Note I couldn't entirely figure out how to get a list of watched files, so there are two watched dependencies tests that need finishing.

:beers:

### CODE REVIEW

**Comment by rhys-vdw** [comment]
Hey @fny, this is awesome! I have to check it out and give it a spin, but looks it looks great. Thanks. :D

File: test.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    expect(stats.compilation.errors).toEqual([])
+    expectInOutput(`var helloWorld = 'Hello World'`)
+    done()
+  })
+})
+
+test('loads with erb', function (done) {
+  compile2({ file: 'engine.js.erb', engine: 'erb' }, done, function (stats) {
+    expect(stats.compilation.errors).toEqual([])
+    expectInOutput('var engine = 'erb'')
```
Comment (rhys-vdw) [inline test.js]: Looks like there's a syntax error here (and in the other similar tests).

```
> rails-erb-loader@5.0.2 test /Users/rhys/Projects/usability-hub/erb-loader
> jest

 FAIL  ./test.js
  ● Test suite failed to run

    /Users/rhys/Projects/usability-hub/erb-loader/test.js: Unexpected token, expected , (66:34)
        64 |   compile2({ file: 'engine.js.erb', engine: 'erb' }, done, function (stats) {
        65 |     expect(stats.compilation.errors).toEqual([])
      > 66 |     expectInOutput('var engine = 'erb'')
           |                                   ^
        67 |     done()
        68 |   })
        69 | })
```

**Comment by rhys-vdw** [comment]
Yo @fny. I tried to run the suite and hit the syntax error reported above. Let me know your plan and if you'd like help with anything. I'm happy to leave out the incomplete tests in preference for getting what you've done so far merged.

btw, not necessary for this PR, but it would be nice to get a test case for a transformer path with a space in it, see #35.

**Comment by rhys-vdw** [comment]
Great. Working well!

There's a few more things to do... I'm just listing these here for posterity. Let me know if you're interested in doing any of these. No pressure, I'm happy to merge now, and then we can move these into different issues to be tackled later by yourself or others.

## Configure ESLint for jest

Add https://www.npmjs.com/package/eslint-plugin-jest and in the test file:

```
/* eslint-env jest/globals */
```

So that we don't get undefined errors for all the globals defined by jest.

## Expand `test` script

1. Run ESLint
1. Run webpack 2 tests
1. Run webpack 3 tests

## Set up TravisCI

I can do this, but the goal would be to set up CI for PRs and such. Should run the `test` command.

## Add tests for watchers

Two tests marked "skipped" in file.

Once these are done we can enable [jest/no-disabled-tests](https://github.com/facebook/jest/blob/master/packages/eslint-plugin-jest/docs/rules/no-disabled-tests.md)

## Add test for runner path with a space in it

See #35 

## Add hooks to npm standard scripts

See [npm scripts](https://docs.npmjs.com/misc/scripts)

- [ ] Run tests on `prepublishOnly` script
- [ ] Run `bundle install` on `postinstall` script so that Ruby dependencies will be installed with npm dependencies

**Comment by rhys-vdw** [comment]
@fny I'll merge this tomorrow and move the above TODO points unless you're interested in doing any more on this PR. Thanks again for your contribution.

**Comment by rhys-vdw** [comment]
These changes have been release in 5.1.0.

See #42 #43 #44 for remaining test issues.

**Comment by fny** [comment]
Thanks much. Sorry I didn't have the time to add anything else to the PR.

Faraz

On Jul 27, 2017 7:36 PM, "Rhys van der Waerden" <notifications@github.com>
wrote:

Merged #41 <https://github.com/usabilityhub/rails-erb-loader/pull/41>.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/usabilityhub/rails-erb-loader/pull/41#event-1182706873>,
or mute the thread

**Comment by rhys-vdw** [comment]
All good. 👍

**Pull request merged**